src/Compilers/Core/MSBuildTask/DiscoverEditorConfigFiles.cs

```diff
@@ -0,0 +1,221 @@
+                        {
+                            case '[':
+                                // Once we're in a section, we don't need to go looking for root = true anymore
+                                return false;
+
+                            case 'r':
+                            case 'R':
+                                // Check for case insensitive 'root = true'
+                                const string root = "root";
+                                const string trueLiteral = "true";
+                                if (matchesCaseInsensitiveAscii(line, charIndex, root))
+                                {
+                                    charIndex += root.Length;
+                                    charIndex = skipSpaces(line, charIndex);
+                                    if (charIndex < line.Length)
+                                    {
+                                        switch (line[charIndex])
+                                        {
+                                            case '=':
+                                            case ':':
+                                                charIndex = skipSpaces(line, charIndex + 1);
+                                                if (matchesCaseInsensitiveAscii(line, charIndex, trueLiteral))
+                                                {
+                                                    charIndex += trueLiteral.Length;
+                                                    charIndex = skipSpaces(line, charIndex);
+                                                    // Is the rest of the line empty or a comment?
+                                                    if (charIndex == line.Length ||
+                                                        line[charIndex] == '#' ||
+                                                        line[charIndex] == ';')
+                                                    {
+                                                        return true;
+                                                    }
+                                                }
+                                                break;
+                                        }
+                                    }
+                                }
+                                break;
+
+                            default:
+                                break;
+                        }
+                    }
+                }
+            }
+
+            return false;
+
+            int skipSpaces(string str, int startIndex)
+            {
+                for (int cur = startIndex; cur < str.Length; cur++)
+                {
+                    if (str[cur] != ' ')
+                    {
+                        return cur;
+                    }
+                }
+
+                return str.Length;
+            }
+
+            bool matchesCaseInsensitiveAscii(string str, int strIndex, string targetLower)
```

Comment:
I'm assuming that the editorconfig spec says that these can only be ascii?

Reply:
Yup